File: 2026/04-April/28-Tuesday/solution.js
var minOperations = function(grid, x) {
    let nums = [];

    for (let row of grid) {
        nums.push(...row);
    }

    nums.sort((a, b) => a - b); 

    let base = nums[0] % x;
    for (let num of nums) {
        if (num % x !== base) return -1;
    }

    let median = nums[Math.floor(nums.length / 2)];
    let ops = 0;

    for (let num of nums) {
        ops += Math.floor(Math.abs(num - median) / x);
    }

    return ops;
};

File: 2026/04-April/28-Tuesday/solution.kt
class Solution {
    fun minOperations(grid: Array<IntArray>, x: Int): Int {
        val nums = mutableListOf<Int>()

        for (row in grid) {
            for (num in row) {
                nums.add(num)
            }
        } 

        nums.sort()

        val base = nums[0] % x
        for (num in nums) {
            if (num % x != base) return -1
        }

        val median = nums[nums.size / 2]
        var ops = 0

        for (num in nums) {
            ops += Math.abs(num - median) / x
        }

        return ops
    }
}
